tests/test_raw.py

```diff
@@ -55,3 +67,47 @@ def test_writeread(self):
         self.assertEqual(parsed['snap'], str(self.snap))
         self.assertEqual(parsed['expid'], str(self.exposure))
         self.assertEqual(parsed['filter'], str(self.filter))
+
+    def test_snap(self):
+        """Test that Snap objects can be constructed from parseable paths.
+        """
+        path = get_raw_path(self.instrument, self.detector, self.group, self.snap, self.exposure, self.filter)
+        parsed = Snap.from_oid(path)
+        self.assertIsNotNone(parsed)
+        # These tests automatically include type-checking.
+        self.assertEqual(parsed.instrument, self.instrument)
+        self.assertEqual(parsed.detector, self.detector)
+        self.assertEqual(parsed.group, self.group)
+        self.assertEqual(parsed.snap, self.snap)
+        self.assertEqual(parsed.exp_id, self.exposure)
+        self.assertEqual(parsed.filter, self.filter)
+
+    def test_bad_snap(self):
+        path = get_raw_path(self.instrument, f"{self.detector}b", self.group,
+                            self.snap, self.exposure, self.filter)
+        with self.assertRaises(ValueError):
```

Comment:
Worth a `assertRaisesRegex` to ensure that you're getting the ValueError you expected?

Reply:
I don't see why the error message matters; what's important is that the non-matching path is detected and can be handled.

Reply:
There may be other things that raise `ValueError`, so I like to ensure that the error was raised at the place I expected it. Easiest way to check that is via the message.